core/tests/protractor/accessibility.js

```diff
@@ -19,28 +19,158 @@
 var waitFor = require('../protractor_utils/waitFor.js');
+var users = require('../protractor_utils/users.js');
+var workflow = require('../protractor_utils/workflow.js');
 
+var AdminPage = require('../protractor_utils/AdminPage.js');
+var CreatorDashboardPage =
+  require('../protractor_utils/CreatorDashboardPage.js');
+var ExplorationEditorPage =
+  require('../protractor_utils/ExplorationEditorPage.js');
+var ExplorationPlayerPage = require(
+  '../protractor_utils/ExplorationPlayerPage.js');
+var LearnerDashboardPage =
+  require('../protractor_utils/LearnerDashboardPage.js');
 var LibraryPage = require('../protractor_utils/LibraryPage.js');
+const { browser } = require('protractor');
+
+var backButton = element(by.css('#backButtonId'));
+var categoryBar = element(by.css(
+  '.protractor-test-search-bar-dropdown-toggle'));
+var continueButton = element(by.css('.protractor-test-continue-button'));
+var mainContent = element(by.css('.protractor-test-main-content'));
+var nextButton = element(by.css('.protractor-test-next-button'));
+var oppiaLogo = element(by.css('.protractor-test-oppia-main-logo'));
+var searchBar = element(by.css('.protractor-test-search-input'));
+var skipLink = element(by.css('.protractor-test-skip-link'));
+
+var EXPLORATION = {
+  title: 'A new exploration',
+  category: 'Learning',
+  objective: 'The goal is to create a new exploration',
+  language: 'English'
+};
 
 describe('screenreader and keyboard user accessibility features', function() {
+  var adminPage = null;
+  var creatorDashboardPage = null;
+  var explorationEditorPage = null;
+  var explorationEditorMainTab = null;
+  var explorationEditorSettingsTab = null;
+  var explorationPlayerPage = null;
   var libraryPage = null;
+  var learnerDashboardPage = null;
 
-  beforeEach(function() {
+
+  beforeAll(function() {
+    adminPage = new AdminPage.AdminPage();
     libraryPage = new LibraryPage.LibraryPage();
+    learnerDashboardPage = new LearnerDashboardPage.LearnerDashboardPage();
+    // The editor and player page objects are only required for desktop testing.
+    if (!browser.isMobile) {
+      creatorDashboardPage = new CreatorDashboardPage.CreatorDashboardPage();
+      explorationEditorPage = new ExplorationEditorPage.ExplorationEditorPage();
+      explorationEditorMainTab = explorationEditorPage.getMainTab();
+      explorationEditorSettingsTab = explorationEditorPage.getSettingsTab();
+      explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
+    }
   });
 
+  var checkActionShortcuts = async function(key, elementToFocus) {
+    // Should move the focus to the elementToFocus.
+    await browser.actions().sendKeys(key).perform();
+    expect(await elementToFocus.getAttribute('id')).toEqual(
+      await (await browser.driver.switchTo().activeElement())
+        .getAttribute('id'));
+
+    // Should move the focus away from the elementToFocus.
+    await browser.actions().sendKeys(protractor.Key.TAB).perform();
```

Comment:
In creating the function, we seem to have lost the check that the IDs of the active element and `elementToFocus` are different after switching focus away. Could we put that check back?